doc/source/data/working-with-text.rst

```diff
@@ -0,0 +1,131 @@
+Working with Text
+=================
+
+Text data is ubiquitous in use cases like Natural Language Processing.
+
+This guide shows you how to:
+
+* :ref:`Read text files <reading-text-files>`
+* :ref:`Transform text <transforming-text>`
+* :ref:`Perform inference on text <performing-inference-on-text>`
+* :ref:`Save text <saving-text>`
+
+.. _reading-text-files:
+
+Reading text files
+------------------
+
+To read text files, call :func:`~ray.data.read_text`. Ray Data creates a row for each
```

Comment:
we also want to show read_binary_files here as well.

Lot's of text data are not cleanly stored in `.txt` files, and can be in `.pdf`, or `.html` or other file formats for example.

I don't think it's obvious `read_binary_files` should be used for these cases

Reply:
even showing `read_json` could be useful. You get structured logs in json format